File: eccorce_app/app/src/main/java/com/example/eccomerce_app/dto/AuthDto.kt
package com.example.eccomerce_app.dto

import kotlinx.serialization.Serializable

@Serializable
data class LoginDto(
    val Username: String,
    val Password: String,
    val DeviceToken: String
)

@Serializable
data class SignupDto(
    val Name: String,
    val Phone: String,
    val Email: String,
    val Password: String,
    val DeviceToken: String
)

@Serializable
data class AuthDto(
    val token: String="",
    val refreshToken: String = ""
)


@Serializable
data class ForgetPasswordDto(val Email: String)


@Serializable
data class CreateVerificationDto(
    val Email:String,
    val Otp: String
)


@Serializable
data class CreateReseatPasswordDto(
    val Email: String,
    val Otp: String,
    val Password: String
)

File: backend/api/Presentation/dto/AuthDto.cs
using System.ComponentModel.DataAnnotations;

namespace api.Presentation.dto
{
    public class AuthDto
    {
        public string Token { get; set; }
        public string RefreshToken { get; set; }
    }

    public enum enRole
    {
       Admin,
       User,
    }

    public class SignupDto
    {
        [StringLength(maximumLength: 50, ErrorMessage = "Name must not be empty")]
        [Required]
        public string Name { get; set; } = string.Empty;

        [StringLength(maximumLength: 13, MinimumLength = 9, ErrorMessage = "phone must between  9 and 13 characters")]
        [Required]
        [MinLength(9, ErrorMessage = "phone must between 9 and 13 characters")]
        [MaxLength(13, ErrorMessage = "phone must between 9 and 13 characters")]
        public string Phone { get; set; } = string.Empty;

        [Required] public string Email { get; set; } = string.Empty;

        [Required] public string Password { get; set; } = string.Empty;

        public string? DeviceToken { get; set; } = null;
        public enRole? Role { get; set; } = enRole.User;
    }


    public class LoginDto
    {
        [StringLength(maximumLength: 50,  ErrorMessage = "username must not be empty")]
        [Required]
        public string Username { get; set; } = string.Empty;

        [Required] public string Password { get; set; } = string.Empty;

        public string? DeviceToken { get; set; } = null;
    }

    public class ForgetPasswordDto
    {
        [Required]
        public string Email { get; set; }=string.Empty;
    }


    public class CreateVerificationDto
    {
        [Required]
        public string Email { get; set; }=String.Empty;
        [Required]
        public string Otp { get; set; }=String.Empty;
    }


    public class CreateReseatePasswordDto
    {
        [Required] public string Email { get; set; }=string.Empty;
        [Required] public string Otp { get; set; } = string.Empty;
        [Required] public string Password { get; set; }=string.Empty;
    }
}